docs/Software Engineering Blog/posts/spring-boot-3-1-docker-compose-support-intuitive-consistent-develop-run-tests-containers.md

```diff
@@ -0,0 +1,239 @@
+                    }
+                }
+            }
+        }
+    }
+  }
+  ```
+
+In this way, the image built by JIB will not include the `spring-boot-docker-compose` library and can run like a normal Spring Boot Application. 
+
+### For Testing
+
+#### Activate Docker in Tests:
+
+In default, the `spring-boot-docker-compose` is disabled when running tests. So we need to activate it by ourselves via property: `spring.docker.compose.skip.in-tests`. Besides, the root folder when running a test is the module folder, so we need to provide the file path to the `root/compose.yaml`. In result, we will write a `test/resources/application.yaml` like:
+
+```yaml
+spring:
+  docker:
+    compose:
+      enabled: true
+      file: ../../compose.yml  # my test class is in root/modules/client
+      skip:
+        in-tests: false
+      profiles:
+        active: ${spring.profiles.include}
+```
+
+Then, in our test class, just add the `@SpringBootTest`, so that the docker-compose will trigger by spring boot app.
+
+#### Shared Containers Across Modules:
+
+If your application consists of multiple modules, sharing Docker containers across tests can significantly speed up the
+testing process. Instead of spinning up new containers for each test module, containers can be shared, reducing the startup time and resource usage.
+
+If we set the `spring.docker.compose.lifecycle-management` as `start-only` ([reference to document](https://docs.spring.io/spring-boot/reference/features/dev-services.html#features.dev-services.docker-compose.lifecycle)), then the docker cluster won't stop after each test. In the second test, which needs to start the docker-compose, will find the docker cluster is already up and ignore that.
+
+![profile_not_updated.png](resources/sb-docker/profile_not_updated.png)
+
+## Summary of Advantages
+
+Using Spring-Boot-Docker-Compose in both development and testing environments provides several key benefits:
+
+### No Prior Knowledge Required:
+
+Developers don't need to know and execute `docker-compose` or `makefile` to run dependency components before running the Spring Boot app. The setup can be standardized and automatic, allowing even newcomers to the project to start the application quickly.
+
+### Decoupling TestContainer Java Library
+
+By using this technic, we can be free from finding the supported TestContainer library for each component, when we need to do extra initialize setup, such as creating schema and data in [Mysql-TestContainer](https://java.testcontainers.org/modules/databases/mysql/), adding mock-rules for [MockServer TestContainer](https://java.testcontainers.org/modules/mockserver/), …etc. We can use the same setting method as we used in developing.
+
+### Test Case Consistency:
+
+By using Spring-Boot-Docker-Compose for tests, we ensure that all test data is consistent between development and test. Reducing the effort to try to make extra test data in developing features.
```

Comment:
📝 **[vale]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[write-good.E-Prime] Try to avoid using 'is'.